File: DSA/Naman-Dixit/Day1/day2/day3/question1.java
import java.util.Scanner;

public class PangramChecker {

  public static String isPangram(String sentence) {
    // Create a boolean array to track encountered letters (a-z)
    boolean[] seenLetters = new boolean[26];

    // Convert sentence to lowercase for case-insensitive check
    sentence = sentence.toLowerCase();

    // Iterate through each character
    for (char c : sentence.toCharArray()) {
      // Check if character is a lowercase letter (a-z)
      if (c >= 'a' && c <= 'z') {
        // Mark the corresponding index in seenLetters as true
        seenLetters[c - 'a'] = true;
      }
    }

    // Check if all letters have been encountered
    for (boolean seen : seenLetters) {
      if (!seen) {
        return "incomplete";
      }
    }

    return "complete";
  }

  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
    System.out.print("Enter a sentence: ");
    String userSentence = scanner.nextLine();

    String result = isPangram(userSentence);
    System.out.println("The sentence is " + result + ".");
  }
}

File: DSA/Devansh-Rawat/Day-3/Question-1.cpp
#include <bits/stdc++.h>

using namespace std;
string isPangram(string sentence)
{
    set<char> unique_alphabet;
    for (char ch : sentence)
    {
        if (isalpha(ch))
        {
            unique_alphabet.insert(tolower(ch));
        }
    }

    if (unique_alphabet.size() == 26)
    {
        return "complete";
    }
    else
    {
        return "incomplete";
    }
}
int main()
{
    string sentence;
    cout << "Enter the sentence: ";
    cin >> sentence;
    cout << isPangram(sentence) << endl;
    return 0;
}